damus/Views/Relays/RelayConfigView.swift

```diff
@@ -9,24 +9,55 @@ import SwiftUI
 
 struct RelayConfigView: View {
     let state: DamusState
+    
     @State var new_relay: String = ""
     @State var relays: [RelayDescriptor]
     @State private var showActionButtons = false
     
+    @StateObject var profile: ProfileModel
+    
     @Environment(\.dismiss) var dismiss
     
-    init(state: DamusState) {
+    init(state: DamusState, profile: ProfileModel) {
         self.state = state
         _relays = State(initialValue: state.pool.our_descriptors)
+        self._profile = StateObject(wrappedValue: profile)
     }
     
-    var recommended: [RelayDescriptor] {
-        let rs: [RelayDescriptor] = []
-        return BOOTSTRAP_RELAYS.reduce(into: rs) { xs, x in
-            if state.pool.get_relay(x) == nil, let url = RelayURL(x) {
-                xs.append(RelayDescriptor(url: url, info: .rw))
+    var recommended: [(RelayDescriptor, Int)] {
```

Comment:
IMO this is too much "smarts" for a view to have. This kind of calculation should be done in a model or helper function, where it can be easily tested. In this case I think `ProfileModel` would be a better place for it.

Reply:
Now implements `func get_recommended_relays() -> [(RelayDescriptor, Int)] {}`